### PR TITLE
Split IConfigurationProvider and IConfigurationPersister implementation

### PR DESCRIPTION
**Pull request by rita-g-sonarsource**
Prep for #1797

### CODE REVIEW

File: src/Integration/MefServices/VsSessionHost.cs [hunk 1]
```diff
@@ -336,8 +336,24 @@ private ILocalService GetConfigurationProvider()
 
             var bindingSerializer = new SolutionBindingSerializer(sccFileSystem, bindingFileLoader, credentialsLoader);
 
-            return new ConfigurationProvider(legacyConfigPathProvider, connectedModeConfigPathProvider, bindingSerializer,
-                legacyConfigFolderItemAdder);
+            return new ConfigurationPersister(legacyConfigPathProvider, connectedModeConfigPathProvider, bindingSerializer, legacyConfigFolderItemAdder);
```
Comment (rita-g-sonarsource) [inline src/Integration/MefServices/VsSessionHost.cs:339]: This copy-paste will go away in future PRs

File: src/Integration/MefServices/VsSessionHost.cs [hunk 2]
```diff
@@ -336,8 +336,24 @@ private ILocalService GetConfigurationProvider()
...
+        }
+
+        private ILocalService GetConfigurationProvider()
```
Comment (rita-g-sonarsource) [inline src/Integration/MefServices/VsSessionHost.cs:342]: GetConfigurationProvider and GetConfigurationPersister are identical as the moment. This copy-paste will go away in future PRs, as I split more dependencies.

File: src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿---
 ################################
 # Assembly references report
-# Report date/time: 2020-10-27T14:54:08.8664584Z
+# Report date/time: 2020-10-29T09:42:23.0046223Z
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt:4]: It doesn't look like anything has changed in the references, which is what I'd expect. However, in that case I wouldn't have expected the AsmRef targets to have updated the baseline in this case.

Committing these unnecessary changes is fine; I'm just confused why the tooling produced them. Did you make reference changes locally then revert them?

**Review by duncanp-sonar — APPROVED** [review]

File: src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt (hunk 3 above)
Comment (rita-g-sonarsource) [reply src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt:4]: Hm, I can't recall. I'll try to notice it if it happens again.
